frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/network-attachment-definition.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { sortable } from '@patternfly/react-table';
+import { ListPage, Table, TableData, TableRow } from '@console/internal/components/factory';
+import { Kebab, ResourceKebab, ResourceLink } from '@console/internal/components/utils';
+import { NamespaceModel } from '@console/internal/models';
+import { K8sResourceKindReference } from '@console/internal/module/k8s';
+import { getName, getNamespace, getUID } from '@console/shared/src';
+import { getConfig } from '../../utils';
+import { NetworkAttachmentDefinitionModel } from '../../models';
+import { NetworkAttachmentDefinitionKind } from '../../types';
+
+const { common } = Kebab.factory;
+const menuActions = [...common];
+
+export const NetworkAttachmentDefinitionReference: K8sResourceKindReference =
+  'NetworkAttachmentDefinition';
+
+const tableColumnClasses = [
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  classNames('col-lg-4', 'col-md-4', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  Kebab.columnClass,
+];
+
+const NetworkAttachmentDefinitionsHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Namespace',
+      sortField: 'metadata.namespace',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Type',
+      sortFunc: 'string',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: '',
+      props: { className: tableColumnClasses[3] },
+    },
+  ];
+};
+NetworkAttachmentDefinitionsHeader.displayName = 'NADTableHeader';
+
+const NetworkAttachmentDefinitionsRow: React.FC<NetworkAttachmentDefinitionsRowProps> = ({
+  obj,
+  index,
+  key,
+  style,
+}) => {
+  const config = getConfig(obj);
+  const name = getName(obj);
+  const namespace = getNamespace(obj);
+  const uid = getUID(obj);
+
+  return (
+    <TableRow id={uid} index={index} trKey={key} style={style}>
+      <TableData className={classNames(tableColumnClasses[0], 'co-break-word')}>
+        <ResourceLink
+          kind={NetworkAttachmentDefinitionModel.kind}
+          name={name}
+          namespace={namespace}
+        />
+      </TableData>
+      <TableData className={classNames(tableColumnClasses[1], 'co-break-word')}>
+        <ResourceLink kind={NamespaceModel.kind} name={namespace} title={namespace} />
+      </TableData>
+      <TableData className={tableColumnClasses[2]}>
+        {config.type || <span className="text-secondary">Not available</span>}
```

Comment:
could you please make a `<NotAvailable />` component?

Reply:
Makes sense. But as this is already used on other multiple places, I suggest to do it consistently in a follow-up.

Reply:
Agreed. I'll make note to take care of this in a follow-up PR.